packages/x-scheduler/src/internals/components/event-popover/content/ReadonlyContent.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { Calendar } from 'lucide-react';
+import { useStore } from '@base-ui-components/utils/store';
+import { CalendarEventOccurrence } from '@mui/x-scheduler-headless/models';
+import { useSchedulerStoreContext } from '@mui/x-scheduler-headless/use-scheduler-store-context';
+import { DEFAULT_EVENT_COLOR } from '@mui/x-scheduler-headless/constants';
+import { selectors } from '@mui/x-scheduler-headless/scheduler-selectors';
+import { useAdapter } from '@mui/x-scheduler-headless/use-adapter';
+import EventPopoverHeader from '../header/EventPopoverHeader';
+import { useTranslations } from '../../../utils/TranslationsContext';
+import { getColorClassName } from '../../../utils/color-utils';
+
+type ReadonlyContentProps = {
+  occurrence: CalendarEventOccurrence;
+  onClose: () => void;
+};
+
+export default function ReadonlyContent(props: ReadonlyContentProps) {
+  const { occurrence, onClose } = props;
+
+  // Context hooks
+  const adapter = useAdapter();
+
+  const translations = useTranslations();
+  const store = useSchedulerStoreContext();
+
+  // Selector hooks
+  const color = useStore(store, selectors.eventColor, occurrence.id);
+  const resource = useStore(store, selectors.resource, occurrence.resource);
+  const defaultRecurrenceKey = useStore(
+    store,
+    selectors.defaultRecurrencePresetKey,
+    occurrence.rrule,
+    occurrence.start,
+  );
+
+  const weekday = adapter.format(occurrence.start, 'weekday');
+  const normalDate = adapter.format(occurrence.start, 'normalDate');
+
+  const recurrenceLabelsMap: Record<string, string> = {
```

Comment:
It's not worth computing the translations for all the presets, to then only use one.

I think the cleanust solution here is to create a utily function that takes the adapter, the start date and the `defaultRecurrenceKey` and uses a `switch` to compute the translation.